Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -124,9 +124,31 @@ class BrowserViewController: UIViewController {
     }
 
     override func traitCollectionDidChange(previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
         updateToolbarState()
         showToolbars(animated: false)
```

Comment:
Ya viewWillTransitionSize and traitCollectionDidChange both get called for almost the same reason. viewWillTransitionSize fires when the screen size has changed which happens on rotation and traitCollectionDidChange gets called when the view controller's size class changes. Since the size class is defined based on an abstract concept involving device + screen size, this also gets called when the screen rotates. I think that using the traitCollection callback makes more sense in this case because we're showing the same UI for the iPhone landscape/iPad portrait views which share the same size class but not the same screen dimensions.